src/applications/proxy-rewrite/tests/e2e/proxy-rewrite-entry.cypress.spec.js

```diff
@@ -0,0 +1,55 @@
+import manifest from '../../manifest.json';
+import { proxyRewriteWhitelist as whitelist } from '../../proxy-rewrite-whitelist.json';
+
+describe(manifest.appName, () => {
+  // Skip tests in CI until the app is released.
+  // Remove this block when the app has a content page in production.
+  before(function() {
+    if (Cypress.env('CI')) this.skip();
+  });
+
+  it('Should not inject header & footer on non-whitelisted host', () => {
+    const hostToEval = 'www.innovation.va.gov';
+    const isWhitelisted = whitelist.some(item => item.hostname === hostToEval);
+
+    if (!isWhitelisted) {
+      cy.visit(`https://${hostToEval}`);
+      cy.get('.consolidated > header').should('not.exist');
+      cy.get('.consolidated > footer').should('not.exist');
+    } else {
+      throw new Error('Evaluated hostname is whitelisted.');
+    }
+  });
+
+  it('Should not inject header & footer on whitelisted hosts that are cookie-only', () => {
+    const hostToEval = 'www.volunteer.va.gov';
+    const listItem = whitelist.find(item => item.hostname === hostToEval);
+    const { cookieOnly } = listItem;
+
+    if (cookieOnly) {
+      cy.visit(`https://${hostToEval}`);
+      cy.get('.consolidated > header').should('not.exist');
+      cy.get('.consolidated > footer').should('not.exist');
+    } else {
+      throw new Error(
+        'Evaluated hostname is not whitelisted or is not cookie-only.',
+      );
+    }
+  });
+
+  it('Should inject header & footer on whitelisted hosts that are not cookie-only', () => {
+    const hostToEval = 'www.benefits.va.gov';
```

Comment:
We used to run Jest snapshot tests to verify injection, but it was removed because they were so unstable (https://github.com/department-of-veterans-affairs/vets-website/commit/6361a11101277b30ee80ca5c42bf108cacd1cbac). We just don't want a hiccup in an external site/service to block a deployment on our side (which this would since it's running in master CI).

It's for sure a good idea to test them, but if we can't mock it like the rest of the API-ish calls I think we should do it in a new github actions workflow that runs on schedule instead.